ext/serializer.c

```diff
@@ -46,6 +46,16 @@ extern void (*profiling_notify_trace_finished)(uint64_t local_root_span_id,
                                                zai_str span_type,
                                                zai_str resource);
 
+static void mpack_write_utf8_lossy_cstr(mpack_writer_t *writer, const char *str, size_t len) {
+    char *utf8 = ddtrace_strip_invalid_utf8(str, &len);
+    if (utf8) {
+        mpack_write_str(writer, utf8, len);
+        ddtrace_drop_rust_string(utf8, len);
+    } else {
+        mpack_write_str(writer, str, len);
```

Comment:
We were doing `mpack_write_cstr` before the change, what's the diff with `mpack_write_str`?

Reply:
cstr just does strlen(str) as length - it's virtually identical.